core/src/main/scala/org/apache/spark/storage/BlockManager.scala

```diff
@@ -1829,7 +1901,63 @@ private[spark] class BlockManager(
     data.dispose()
   }
 
+  /**
+   * Class to handle block manager decommissioning retries
+   * It creates a Thread to retry offloading all RDD cache blocks
+   */
+  private class BlockManagerDecommissionManager(conf: SparkConf) {
+    @volatile private var stopped = false
+    private val sleepInterval = conf.get(
+      config.STORAGE_DECOMMISSION_REPLICATION_REATTEMPT_INTERVAL)
+
+    private val blockReplicationThread = new Thread {
+      override def run(): Unit = {
+        var failures = 0
+        try {
+          while (blockManagerDecommissioning &&
+            !stopped &&
+            !Thread.interrupted() &&
```

Comment:
When the thread is Interrupted, the InterruptedException is caught and stopped is set to true. So do we need to add `!Thread.interrupted()` explicitly? Or is this just for being more defensive here?

Reply:
This is incase one of our sub functions is capturing the thread interruption (either now or in the future).

Reply:
got it.